command.js

```diff
@@ -0,0 +1,117 @@
+var exec = require('child_process').exec;
+var Promise = require('promise');
+var debug = require('debug')('mozdevice:command');
+
+/**
+ * API for building up and executing shell commands
+ * @param {string} [initialCommand] An optional first command
+ * @constructor
+ */
+var Command = function(initialCommand) {
+  this.builder = initialCommand ? [initialCommand] : [];
+};
+
+/**
+ * Append content to the command builder
+ * @param {string} content
+ * @returns {Command}
+ */
+Command.prototype.append = function(content) {
+  this.builder.push(content);
+  return this;
+};
+
+/**
+ * Append a command to be run against `adb <command>`
+ * @param {string} command
+ * @returns {Command}
+ */
+Command.prototype.adb = function(command) {
+  this.builder.push('adb ' + command);
+  return this;
+};
+
+/**
+ * Append a command to be run against `adb shell <command>`
+ * @param {string} command
+ * @returns {Command}
+ */
+Command.prototype.adbShell = function(command) {
+  this.builder.push("adb shell '" + command + "'");
```

Comment:
Because sometimes you'll use something that just doesn't work, like:

`adb shell stop b2g && start b2g`

Putting it in quotes lets adb shell run the whole command remotely:

`adb shell 'stop b2g && start b2g'`